### PR TITLE
VIH-6672 VHO_sees_Witness_dismissed_alert 

### PR DESCRIPTION
**Pull request by RocheL**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/VIH-6672
### Change description ###
 - added endpoint to add a task (alert) for a participant in a conference
**Does this PR introduce a breaking change?** (check one with "x")
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: VideoAPI/Video.API/Validations/AddTaskRequestValidation.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+namespace Video.API.Validations
+{
+    public class AddTaskRequestValidation : AbstractValidator<AddTaskRequest>
+    {
+        public const string NoTaskTypeError = "Task type is required";
+
+        public AddTaskRequestValidation()
+        {
+            RuleFor(x => x.TaskType).IsInEnum().WithMessage(NoTaskTypeError);
```
Comment (shaed-parkar) [inline VideoAPI/Video.API/Validations/AddTaskRequestValidation.cs:12]: Would you validate the body is not empty either please?

File: VideoAPI/VideoApi.UnitTests/Controllers/Tasks/TasksControllerTests.cs [hunk 2]
```diff
@@ -68,5 +68,28 @@ public async Task Should_return_not_found_when_no_task_is_found()
             typedResult.StatusCode.Should().Be((int)HttpStatusCode.NotFound);
             commandHandler.Verify(c => c.Handle(It.IsAny<UpdateTaskCommand>()), Times.Once);
         }
+
+        [Test]
+        public async Task Should_return_no_content_on_add_a_task_for_the_participant_in_a_conference()
```
Comment (shaed-parkar) [inline VideoAPI/VideoApi.UnitTests/Controllers/Tasks/TasksControllerTests.cs:73]: I don't think we need to test with mocks when you're testing with the API already in the integration project.

**Review by shaed-parkar — COMMENTED** [review]
Pretty good, just a few suggestions

File: VideoAPI/Testing.Common/Helper/ApiUriFactory.cs [hunk 3]
```diff
@@ -84,6 +84,7 @@ public static class TaskEndpoints
             private const string ApiRoot = "conferences";
             public static string GetTasks(Guid conferenceId) => $"{ApiRoot}/{conferenceId}/tasks";
             public static string UpdateTaskStatus(Guid conferenceId, long taskId) => $"{ApiRoot}/{conferenceId}/tasks/{taskId}";
+            public static string AddTask(Guid conferenceId, Guid participantId) => $"{ApiRoot}/{conferenceId}/participant/{participantId}/task";
```
[the remark this answers is not in the document]
Comment (RocheL) [reply VideoAPI/Testing.Common/Helper/ApiUriFactory.cs]: removed participant

File: VideoAPI/Video.API/Controllers/TasksController.cs [hunk 4]
```diff
@@ -99,5 +99,34 @@ public async Task<IActionResult> GetTasksForConferenceAsync(Guid conferenceId)
...
+        /// </summary>
+        /// <param name="conferenceId">The conference Id to add the task</param>
+        /// <param name="participantId">The participant Id to add the task for.</param>
+        /// <param name="addTaskRequest">The task request containing the task type and the alert (task name)</param>
+        /// <returns></returns>
+        [HttpPost("{conferenceId}/participant/{participantId}/task")]
+        [SwaggerOperation(OperationId = "AddTask")]
+        [ProducesResponseType((int)HttpStatusCode.BadRequest)]
+        [ProducesResponseType((int)HttpStatusCode.NoContent)]
+        public async Task<IActionResult> AddTaskAsync(Guid conferenceId, Guid participantId, [FromBody] AddTaskRequest addTaskRequest)
```
[the remark this answers is not in the document]
Comment (RocheL) [reply VideoAPI/Video.API/Controllers/TasksController.cs]: moved participant to the request model

File: VideoAPI/Video.API/Controllers/TasksController.cs [hunk 5]
```diff
@@ -99,5 +99,34 @@ public async Task<IActionResult> GetTasksForConferenceAsync(Guid conferenceId)
...
+            try
+            {
+                var command = new AddTaskCommand(conferenceId, participantId, addTaskRequest.Body, addTaskRequest.TaskType);
+                await _commandHandler.Handle(command);
+
+                return NoContent();
+            }
+            catch (Exception e)
+            {
+                _logger.LogError(e, $"Unable to add a task {addTaskRequest.Body} for participant {participantId} in conference {conferenceId}");
```
[the remark this answers is not in the document]
Comment (RocheL) [reply VideoAPI/Video.API/Controllers/TasksController.cs]: update the logging code

File: VideoAPI/Video.API/Validations/AddTaskRequestValidation.cs (hunk 1 above)
Comment (RocheL) [reply VideoAPI/Video.API/Validations/AddTaskRequestValidation.cs:16]: added validation for participant and body

File: VideoAPI/VideoApi.UnitTests/Controllers/Tasks/TasksControllerTests.cs (hunk 2 above)
Comment (RocheL) [reply VideoAPI/VideoApi.UnitTests/Controllers/Tasks/TasksControllerTests.cs]: removed the tests

**Review by shaed-parkar — APPROVED** [review]
LGTM

**Pull request merged**